projects/frontend/shared-components/gui/projects/shared/src/lib/core/navigation/services/navigation.service.ts

```diff
@@ -0,0 +1,351 @@
+
+
+/*
+ * Copyright 2021 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { ActivatedRoute, ActivatedRouteSnapshot, NavigationExtras, Router } from '@angular/router';
+
+import { take } from 'rxjs/operators';
+
+import { ArrayElement, CollectionsUtil } from '../../../utils';
+
+import { Replacer, TaurusNavigateAction } from '../../../common';
+
+import { SE_NAVIGATE, SystemEventHandler, SystemEventHandlerClass } from '../../system-events';
+
+import { RouterService, RouteState } from '../../router';
+import { RouteStateFactory } from '../../router/factory';
+
+/**
+ * ** Service should be provided from the Root injector in Application.
+ *
+ *   - e.g. TMS Control Plane or SuperCollider Control Plane, etc..
```

Comment:
Please, remove TMS, SuperCollider and Control Plane occurrences from the code base. 

All these are VMW internals.